src/storage/src/table/cell_based_table.rs

```diff
@@ -359,45 +420,97 @@ impl<S: PkAndRowStream + Unpin> TableIter for S {
 }
 
 /// Iterators
-impl<S: StateStore> CellBasedTable<S> {
+impl<S: StateStore, const T: AccessType> CellBasedTable<S, T> {
+    /// Get multiple [`CellBasedIter`] based on the specified vnodes.
+    async fn iters_with_encoded_key_range<R, B, const ITER_TYPE: bool>(
+        &self,
+        encoded_key_range: R,
+        epoch: u64,
+    ) -> StorageResult<Vec<CellBasedIter<S, ITER_TYPE>>>
+    where
+        R: RangeBounds<B> + Send + Clone,
+        B: AsRef<[u8]> + Send,
+    {
+        // For each vnode, construct an iterator.
+        // TODO: if there're some vnodes continuously in the range and we don't care about order, we
+        // can use a single iterator.
+        let mut iterators = Vec::with_capacity(self.vnodes.num_high_bits());
+
+        for vnode in self
+            .vnodes
+            .iter()
+            .enumerate()
+            .filter(|&(_, set)| set)
+            .map(|(i, _)| i as VirtualNode)
+        {
+            let raw_key_range = prefixed_range(encoded_key_range.clone(), &vnode.to_be_bytes());
+            let iter = CellBasedIter::<_, ITER_TYPE>::new(
+                &self.keyspace,
+                self.mapping.clone(),
+                raw_key_range,
+                epoch,
+            )
+            .await?;
+            iterators.push(iter);
+        }
+
+        Ok(iterators)
+    }
+
     /// Get a [`StreamingIter`] with given `encoded_key_range`.
     pub(super) async fn streaming_iter_with_encoded_key_range<R, B>(
         &self,
         encoded_key_range: R,
         epoch: u64,
     ) -> StorageResult<StreamingIter<S>>
     where
-        R: RangeBounds<B> + Send,
+        R: RangeBounds<B> + Send + Clone,
```

Comment:
Just curious: why add this `clone`?

Reply:
We may construct multiple iterators if there're multiple vnodes, so we need to clone the `range` for all of them. We may let the caller to pass an `Arc` to avoid allocations.